scene/main/viewport.cpp

```diff
@@ -1777,7 +1777,7 @@ Control *Viewport::_gui_find_control_at_pos(CanvasItem *p_node, const Point2 &p_
 	matrix.affine_invert();
 
 	//conditions for considering this as a valid control for return
-	if (c->data.mouse_filter != Control::MOUSE_FILTER_IGNORE && c->has_point(matrix.xform(p_global)) && (!gui.drag_preview || (c != gui.drag_preview && !gui.drag_preview->is_a_parent_of(c)))) {
+	if (c->data.mouse_filter != Control::MOUSE_FILTER_IGNORE && c->has_point(matrix.xform(p_global)) && (!gui.drag_preview_id.is_valid() || (c->get_instance_id() != gui.drag_preview_id && !_gui_get_drag_preview()->is_a_parent_of(c)))) {
```

Comment:
```
!_gui_get_drag_preview()->is_a_parent_of(c)
```
^ this looks like it might still crash when `_gui_get_drag_preview()` returns null.

Also I wonder if this condition could be simpler/better formatted, as it's very long.

Reply:
Yep, that was an oversight on my part. I have refactored that monstrosity into something more readable.